sw/device/lib/testing/fault_handler_testutils.c

```diff
@@ -0,0 +1,45 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/csr.h"
+#include "sw/device/lib/runtime/log.h"
+
+static uint32_t get_mepc(void) {
+  uint32_t mepc;
+  CSR_READ(CSR_REG_MEPC, &mepc);
+  return mepc;
+}
+
+static void set_mepc(uint32_t mepc) { CSR_WRITE(CSR_REG_MEPC, mepc); }
+
+void fault_handler_testutils_set_return_addr(uint32_t *indirection) {
```

Comment:
Since this function will not be necessary when we migrate tests to the OTTF, since it is already (partially) implemented by the OTTF, it may be better to move this to be a static helper function in the test below that uses it.

When we migrate to the OTTF, you  will probably just override the `ottf_instr_access_fault()` function in the test directly, as I describe in my comment below.